tool/tctl/common/auth_command.go

```diff
@@ -618,6 +631,18 @@ func (a *AuthCommand) generateUserKeys(clusterAPI auth.ClientI) error {
 			SessionID:   appSession.GetName(),
 		}
 		certUsage = proto.UserCertsRequest_App
+	case a.dbName != "":
+		server, err := getDatabaseServer(context.TODO(), clusterAPI, a.dbName)
```

Comment:
Right now we are using `--db-name` to provide the name od database instance where in `tsh login/connect` the `--db-name` refers to a database name scheme. For example: 
```	
tsh db connect  --db-user=postgres --db-name=testdb postgres-instance
```
This might be unintuitive also in current approach also the   `Database` field is not set in:

```
routeToDatabase = proto.RouteToDatabase{
			ServiceName: a.dbName,
			Protocol:    server.GetDatabase().GetProtocol(),
			Username:    a.dbUser,
		}
```
so there is no way to specify the --db-name to connect in case of postgres access where databaseName is used.

Reply:
That's an interesting point, but it is hard to name the `ServiceName` property into something else since in some places (like the UI and the command you mentioned), it is called "database name".

What about setting the flags like this:
- `--db`: Refers to the `ServiceName` property from `RouteToDatabase`;
- `--db-name`: Refers to the `Database` property from `RouteToDatabase`;
- `--db-user`: Refers to the `Username` property from `RouteToDatabase`;

Another question related to this: is the parameter (`Database`) required only in some database types? if so should we add validation to it (for example, if PostgreSQL requires it, we make the flag required)?

Reply:
Agree, the `--db-name` is a bit misleading because and it should be somehow distinguishable from  `RouteToDatabase.ServiceName` a new flog `--db` or `--db-service` should fix this and  to allow to provide full `RouteToDatabase` information 

@r0mant Any objection  ?

Reply:
Sorry for delay here guys, yeah I think `--db-service`, `--db-user` and `--db-name` makes sense.

Reply:
Done. Updated the flags.